consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -0,0 +1,765 @@
+            mock_wait_certificate
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_consensus_state.consensus_state_for_block_id.return_value = \
+            mock_state
+
+        mock_consensus_state_store.return_value.__getitem__.return_value = \
+            mock_consensus_state
+
+        # create mock_batch_publisher
+        mock_batch_publisher = mock.Mock(
+            identity_signing_key=signing.generate_privkey())
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_block_header with the following fields
+        mock_block = mock.Mock(identifier='0123456789abcdefedcba9876543210')
+        mock_block.header.signer_pubkey = '90834587139405781349807435098745'
+        mock_block.header.previous_block_id = '2'
+        mock_block.header.block_num = 1
+        mock_block.header.state_root_hash = '6'
+        mock_block.header.batch_ids = '4'
+
+        # create mock_signup_info
+        mock_signup_info.create_signup_info.return_value = \
+            mock.Mock(
+                poet_public_key='poet public key',
+                proof_data='proof data',
+                anti_sybil_id='anti-sybil ID',
+                sealed_signup_data='sealed signup data')
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                        'LOGGER') as mock_logger:
+            block_publisher = \
+                poet_block_publisher.PoetBlockPublisher(
+                    block_cache=mock_block_cache,
+                    state_view_factory=mock_state_view_factory,
+                    batch_publisher=mock_batch_publisher,
+                    data_dir=self._temp_dir,
+                    validator_id='validator_deadbeef')
+
+            self.assertFalse(
+                block_publisher.initialize_block(
+                    block_header=mock_block.header))
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.debug.call_args
+            self.assertTrue('No public key found, so going '
+                            'to register new signup information' in message)
```

Comment:
Additional things you might want to test to verify that the block publisher did the right thing is that the `send` method was called on your mock batch publisher and that `create_signup_info` was called on your mock signup info object.